src/status_im/contexts/wallet/wallet_connect/processing_events.cljs

```diff
@@ -60,44 +67,52 @@
                      assoc
                      :address      (string/lower-case address)
                      :raw-data     raw-data
-                     :display-data (or parsed-data raw-data))})))
+                     :display-data (or parsed-data raw-data))
+      :fx [[:dispatch [:wallet-connect/show-request-modal]]]})))
 
 (rf/reg-event-fx
- :wallet-connect/process-eth-send-transaction
- (fn [{:keys [db]}]
-   (let [event                 (wallet-connect-core/get-db-current-request-event db)
-         display-data          (-> event
-                                   clj->js
-                                   (js/JSON.stringify nil 2))
-
-         {:keys [from] :as tx} (-> event wallet-connect-core/get-request-params first)
-         chain-id              (-> event
-                                   (get-in [:params :chainId])
-                                   wallet-connect-core/eip155->chain-id)]
+ :wallet-connect/prepare-transaction-success
+ (fn [{:keys [db]} [prepared-tx chain-id]]
+   (let [{:keys [tx-args]} prepared-tx
+         tx                (bean/->clj tx-args)
```

Comment:
Did you guys benchmark the code using bean? Is it worth it in events? I see this is growing more in usage, is there a place where we discussed this direction? Sounds good in theory, particularly in events where we will discard the bean.

But I do see a conversion from bean to clj in a sub, for example `:wallet-connect/transaction-args`, which can be worse than paying the cost of conversion once, instead of on every run of the subscription when it's cache is recreated.

cc @ulisesmac

Reply:
Yeah, the conversion inside the sub felt icky to begin with. The point was that we need the transaction in js when sending it, so I nudged towards not converting it to clj and back to js at the end of the flow, but seems like it's necessary.